projects/netgrif-components-core/src/lib/authorization/role/role-guard.service.ts

```diff
@@ -39,31 +39,28 @@ export class RoleGuardService implements CanActivate {
     public canAccessView(view: View, url: string): boolean {
         if (typeof view.access !== 'string' && (view.access.hasOwnProperty('role') || view.access.hasOwnProperty('bannedRole'))) {
 
-            if (view.access.hasOwnProperty('bannedRole')) {
+            if (view.access.hasOwnProperty('role') && view.access.hasOwnProperty('bannedRole')) {
                 const bannedRoles = this.parseRoleConstraints(view.access.bannedRole, url);
+                const allowedRoles = this.parseRoleConstraints(view.access.role, url);
+                return allowedRoles.some(constraint => {return this.decideAccessByRole(constraint)}) &&
```

Comment:
I believe this would be more readable as follows:
```
if (bannedRoles.some(role => this.decideAccessByRole(role)) {
    return false; // bannedRoles are non empty and the user has one of them => no access
}
return allowedRoles.some(role => this.decideAccessByRole(role)); // user was not denied access by a banned role, they need at least one allowed role
```